File: C/caesar_cipher.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <math.h>
#include <ctype.h>
#include <stdlib.h>

int main(int argc, string argv[])
{       
    while (argc != 2)
    {
        printf("Usage: ./caesar.key\n");
        return 1;
    }
    const int K = 1;
    for (int i = 0; i < argc; i++)
        {
           for (int j = 0, n = strlen(argv[K]); j < n; j++) 
           {
               if (isdigit(argv[K][j]))
               {
                    int key = atoi(argv[K]);
                    string plain = get_string("plaintext: ");
                    printf("ciphertext: ");
                    for (int l = 0, m = strlen(plain); l < m; l++)
                    {
                        char c = (char) plain[l];
                        if (isupper(c))
                        {
                            printf("%c",(((c - 'A' + key) % 26) + 'A'));
                        }    
                        else if (islower(c))
                        {
                            printf("%c",(((c - 'a' + key) % 26) + 'a'));
                        }
                        else
                        {
                            printf("%c", c);
                        }
                    }
                    printf("\n");
                    return 0; 
               }
               else
               {
                   printf("Usage: ./caesar.key\n");
                   return 1;
               }
            }
        }
}

File: Python/caesar_cipher.py
from cs50 import get_string
import sys

def main():

    # Checks for argument count, returns 1 if argument count is not 2
    if len(sys.argv) != 2:
        print("Usage: python caesar.py key")
        exit(1)

    # Makes key by turning second argument into an integer
    key = int(sys.argv[1])

    # Encryption Function
    def encrypt(text, key):
        cipher = ""
        for i in range(len(text)):
            char = text[i]

            if (char.isupper()):
                cipher += chr((ord(char) + key - 65) % 26 + 65)
            else:
                cipher += chr((ord(char) + key - 97) % 26 + 97)
            return cipher

    while True:
        plaintext = get_string("Plaintext: ")
        print("ciphertext: ", end = "")
        for c in plaintext:
            if c.isalpha():
                c = encrypt(c, key)
            print(c, end = "")
        print()
        break

if __name__ == "__main__":
    main()